specs/phase1/beacon-chain.md

```diff
@@ -166,246 +91,74 @@ class AttestationData(Container):
-    # Slashings
-    slashings: Vector[Gwei, EPOCHS_PER_SLASHINGS_VECTOR]  # Per-epoch sums of slashed effective balances
-    # Attestations
-    previous_epoch_attestations: List[PendingAttestation, MAX_ATTESTATIONS * SLOTS_PER_EPOCH]
-    current_epoch_attestations: List[PendingAttestation, MAX_ATTESTATIONS * SLOTS_PER_EPOCH]
-    # Finality
-    justification_bits: Bitvector[JUSTIFICATION_BITS_LENGTH]  # Bit set for every recent justified epoch
-    previous_justified_checkpoint: Checkpoint  # Previous epoch snapshot
-    current_justified_checkpoint: Checkpoint
-    finalized_checkpoint: Checkpoint
-    # Phase 1
+class BeaconState(phase0.BeaconState):
+    current_epoch_pending_headers: List[PendingHeader, MAX_PENDING_HEADERS * SLOTS_PER_EPOCH]
+    previous_epoch_pending_headers: List[PendingHeader, MAX_PENDING_HEADERS * SLOTS_PER_EPOCH]
+    most_recent_confirmed_commitments: Vector[Vector[DataCommitment, SLOTS_PER_EPOCH], MAX_SHARDS]
+    shard_gasprice: uint64
     current_epoch_start_shard: Shard
-    shard_states: List[ShardState, MAX_SHARDS]
-    online_countdown: List[OnlineEpochs, VALIDATOR_REGISTRY_LIMIT]  # not a raw byte array, considered its large size.
-    current_light_committee: CompactCommittee
-    next_light_committee: CompactCommittee
-    # Custody game
-    # Future derived secrets already exposed; contains the indices of the exposed validator
-    # at RANDAO reveal period % EARLY_DERIVED_SECRET_PENALTY_MAX_FUTURE_EPOCHS
-    exposed_derived_secrets: Vector[List[ValidatorIndex, MAX_EARLY_DERIVED_SECRET_REVEALS * SLOTS_PER_EPOCH],
-                                    EARLY_DERIVED_SECRET_PENALTY_MAX_FUTURE_EPOCHS]
-    custody_chunk_challenge_records: List[CustodyChunkChallengeRecord, MAX_CUSTODY_CHUNK_CHALLENGE_RECORDS]
-    custody_chunk_challenge_index: uint64
 ```
 
 ## New containers
 
 The following containers are new in Phase 1.
 
-### `ShardBlock`
+### `DataCommitment`
 
 ```python
-class ShardBlock(Container):
-    shard_parent_root: Root
-    beacon_parent_root: Root
-    slot: Slot
-    shard: Shard
-    proposer_index: ValidatorIndex
-    body: ByteList[MAX_SHARD_BLOCK_SIZE]
+class DataCommitment(Container):
+    # Kate commitment to the data
+    point: BLSCommitment
+    # Length of the data in samples
+    length: length
 ```
 
-### `SignedShardBlock`
+### `ShardHeader`
 
 ```python
-class SignedShardBlock(Container):
-    message: ShardBlock
-    signature: BLSSignature
-```
-
-### `ShardBlockHeader`
-
-```python
-class ShardBlockHeader(Container):
-    shard_parent_root: Root
-    beacon_parent_root: Root
+class ShardHeader(Container):
+    # Slot and shard that this header is intended for
     slot: Slot
     shard: Shard
-    proposer_index: ValidatorIndex
-    body_root: Root
-```
-
-### `ShardState`
-
-```python
-class ShardState(Container):
-    slot: Slot
-    gasprice: Gwei
-    latest_block_root: Root
+    # The actual data commitment
+    commitment: DataCommitment
```

Comment:
Are we now doing only a single commitment per block? I thought we would split in two if it runs over the 256 kB?

Reply:
I'm inclined to say keep the single commitment for simplicity for now? But I'm happy to discuss this more.